underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystem.java

```diff
@@ -525,7 +524,7 @@ private boolean isFolder(String key) {
         S3Object[] objs = mClient.listObjects(mBucketName, path, "");
         // If there are, this is a folder and we can create the necessary metadata
         if (objs.length > 0) {
-          mkdirsInternal(path);
+          // TODO(binfan): Enable optimization "mkdirsInternal(path);"
```

Comment:
this is an optimization to convert an S3 encoded dir into an Alluxio encoded dir, which requires write on reading a S3 bucket as ufs. This is introduced as an optimization after 1.1.0. After discussion with Jiri, we think this might have side-effect (e.g., fail to list on a read-only bucket), and let's hide it for this patch release.